src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -827,4 +878,34 @@ private static string EnsureTrailingCharacter(string path, char trailingCharacte
             return path + trailingCharacter;
         }
     }
+
+    class Preconditions
```

Comment:
This method of tracking preconditions makes it impossible to do anything but print them. I would prefer to see this factored in such a way that each precondition type is represented by an enum or class, with appropriate fields for each.